contracts/Market.sol

```diff
@@ -277,32 +277,48 @@ contract Market is Ownable, ERC721 {
 		_settle(index, result);
 	}
 
-	function _settle(uint256 id, bool result) private {
+	function settleMarket(bytes16 marketId) external {
+		_settleMarketByRange(marketId, 0, _marketBets[marketId].length - 1);
+	}
+
+	function settleMarketByRange(bytes16 marketId, uint256 from, uint256 to) external {
+		_settleMarketByRange(marketId, from, to);	
+	}
+
+	function _settleMarketByRange(bytes16 marketId, uint256 from, uint256 to) private {
+		assert(from <= _marketBets[marketId].length);
+		assert(to <= _marketBets[marketId].length);
```

Comment:
best to not use assert for errors that can happen from user input. Instead, it is preferable to use require

Reply:
Disagree.  Asserts should be used in a condition that shouldnt happen

Reply:
https://dev.to/tawseef/require-vs-assert-in-solidity-5e9d